icu4c/source/test/intltest/intltest.cpp

```diff
@@ -59,7 +59,11 @@
 static char* _testDataPath=nullptr;
 
 // Static list of errors found
-static UnicodeString errorList;
+static UnicodeString& getErrorList() {
+    static UnicodeString* errorList = new UnicodeString();
```

Comment:
I am surprised that we don't get a memory leak complaint...

I guess that if the tests pass we don't call this. If they fail, we should get both the test failure and a leak report.